ext/jruby/org/jruby/ext/strscan/RubyStringScanner.java

```diff
@@ -901,9 +901,14 @@ public IRubyObject captures(ThreadContext context) {
         newAry = RubyArray.newArray(runtime, numRegs);
 
         for (i = 1; i < numRegs; i++) {
-            IRubyObject str = extractRange(runtime,
+            IRubyObject str;
+            if (REGION_ADAPTER.getBeg(regs, i) == -1) {
+                str = context.nil;
+            } else {
+                str = extractRange(runtime,
                     adjustRegisterPosition(REGION_ADAPTER.getBeg(regs, i)),
                     adjustRegisterPosition(REGION_ADAPTER.getEnd(regs, i)));
```

Comment:
@headius Should we indent them?

Reply:
I'll merge this. If we need to fix style here, we can do it as a follow-up PR.

Reply:
We generally just try to match existing style when making other changes. We can do a separate PR to clean up formatting.